portfolio/src/main/webapp/script.js

```diff
@@ -12,56 +12,77 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
+
+const loc = "/images/life/";
+let currImageNum = 1;
+let sliderSpeed=5000;
+
 // Function that shows or hides the list of courses
 // for a particular semester upon user click
 function toggleCourseList(listObject) {
     let semester = listObject.id;
     let semCourseList = "courselist" + semester;
     let lst = document.getElementById(semCourseList);
-    if(lst.style.display === "none" || lst.style.display === "")
+    if(listObject.className === "up")
     {
         lst.style.display = "block";
+        document.getElementById(semester).className = "down";
     } else {
         lst.style.display = "none";
+        document.getElementById(semester).className = "up";
     }
 }
 
-// gets the number of the image currently being 
-// displayed on the slideshow
-function getImageNum() {
-    let currImg = document.getElementById("galleryimg");
-    let imgName = currImg.src;
-    const loc = "/images/life/"
-    let startOfImgNum = imgName.indexOf(loc) + loc.length;
-    let endOfImgNum = imgName.indexOf(".jpg");
-    let imgNum = Number(imgName.slice(startOfImgNum, endOfImgNum));
-    return imgNum;
-}
 
 // function that changes slideshow's picture to the next 
 // picture when the user presses the next button
 function next() {
-    const loc = "/images/life/"
-    let imgNum = getImageNum();
-    let newImageNum = (imgNum + 1);
-    if(newImageNum === 17)
+    currImageNum++;
+    if(currImageNum === 17)
     {
-        newImageNum = 1;
+        currImageNum = 1;
     }
-    let newImgPath = loc + newImageNum + ".jpg";
+    clearInterval(myTimer);
+    let newImgPath = loc + currImageNum + ".jpg";
     document.getElementById("galleryimg").src = newImgPath;
+    myTimer = setInterval(loopOverImages, sliderSpeed);
 }
 
 // function that changes slideshow's picture to the previous 
 // picture when the user presses the previous button
 function prev() {
-    const loc = "/images/life/"
-    let imgNum = getImageNum();
-    let newImageNum = (imgNum - 1);
-    if(newImageNum === 0)
+    currImageNum--;
+    if(currImageNum === 0)
     {
-        newImageNum = 16;
+        currImageNum = 16;
     }
-    let newImgPath = loc + newImageNum + ".jpg";
+    clearInterval(myTimer);
+    let newImgPath = loc + currImageNum + ".jpg";
     document.getElementById("galleryimg").src = newImgPath;
+    myTimer = setInterval(loopOverImages, sliderSpeed);
+}
+
+// function that is called every sliderSpeed seconds
+// changes slideshow to display next image in list of 
+// images
+function loopOverImages() 
+{
+    currImageNum++;
+    if(currImageNum === 17)
+    {
+        currImageNum = 1;
+    }
+    let newImgPath = loc + currImageNum + ".jpg";
+    document.getElementById("galleryimg").src = newImgPath;
+}
+myTimer = setInterval(loopOverImages, sliderSpeed);
```

Comment:
nit: Move to the top of the file (with the other global variables) and declare this with `let`.
https://google.github.io/styleguide/jsguide.html#features-use-const-and-let